Sources/Kitura/CodableRouter.swift

```diff
@@ -600,14 +603,31 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
+        switch (identifierExpected, identifierSupplied) {
+        case (false, true):
+            Log.error("Path '\(path)' is not allowed: Codable routes do not allow path parameters.")
+            return false
+        case (true, false):
+            Log.warning("Identifier expected for '\(path)'. :id will be automatically appended to the path.")
+            return true
+        case (true, true):
+            /// Precondition: Path is known to contain a : character
+            func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
+                let parameterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+                let parameter = parameterString.count > 0 ? parameterString[1] : ""
+                return parameter.isEmpty || parameter == "id"
```

Comment:
What happens if the path is `/hello/:`, i.e. it ends with a colon? If I am reading the code correctly the function will return true which doesn't seem right? Please add a test for a route that ends with `:`.